.github/workflows/ci-tests.yml

```diff
@@ -0,0 +1,137 @@
+name: Go Test Workflow
+
+on:
+  push:
+    branches: [master]
```

Comment:
Runs on master will not block bad code nor are they visible to developers. You have to navigate to the Actions tab to see these runs. AIUI, code lands on master only via PRs so this could be removed. 
Also, on L59 below, it assumes PR context.

Reply:
now it will run when targetting master and release branches